DataFormats/Detectors/CTP/src/Configuration.cxx

```diff
@@ -1153,6 +1177,48 @@ int CTPInputsConfiguration::getInputIndexFromName(std::string& name)
   return 0xff;
 }
 
+CtpCfg CtpCfg::readAndSave(std::string& path, int& ret)
+{
+  ret = 0;
+  std::string file = path + filename;
+  std::ifstream ctpcfg(file);
+  if (ctpcfg.is_open()) {
+    std::string line;
+    while (std::getline(ctpcfg, line)) {
+      o2::utils::Str::trim(line);
+      if (line.size() == 0) {
+        continue;
+      }
+      if (line[0] == '#') {
+        continue;
+      }
+      std::vector<std::string> tokens = o2::utils::Str::tokenize(line, ' ');
+      size_t ntokens = tokens.size();
+      if (ntokens < 2) {
+        LOG(warn) << "Not enough tokens";
+        continue;
+      }
+      if (tokens[0].find("TForbits") != std::string::npos) {
+        TFOrbits = std::atol(tokens[1].c_str());
+      } else if (tokens[0].find("ccdb") != std::string::npos) {
+        ccdb = std::atoi(tokens[1].c_str());
+      } else if (tokens[0].find("orbitshift") != std::string::npos) {
+        orbitShift = std::atol(tokens[1].c_str());
+      } else if (tokens[0].find("ir_inputs") != std::string::npos) {
+        irInputs_1_24 = std::stoul(tokens[1].c_str(), nullptr, 16);
+        irInputs_25_48 = std::stoul(tokens[2].c_str(), nullptr, 16);
+      } else {
+        LOG(warn) << " Token not found:" << tokens[0];
+      }
+    }
+    LOG(warn) << "Open file success:" << file;
+  } else {
+    LOG(warn) << "Can not open file:" << file;
+    ret = 1;
+  }
+  return *this;
```

Comment:
@lietava to be sure this is what you need: here you create and return a copy of the object

Reply:
Yes, every call of readandsave was creating on return a new copy of itself.